src/Microsoft.ML.ImageAnalytics/ExtensionsCatalog.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.ImageAnalytics;
+using Microsoft.ML.Transforms;
+
+namespace Microsoft.ML
+{
+    public static class ImageEstimatorsCatalog
+    {
+        /// <summary>
+        /// Converts the images to grayscale.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="columns">The name of the columns containing the image paths(first item of the tuple), and the name of the resulting output column (second item of the tuple).</param>
+        public static ImageGrayscalingEstimator Grayscale(this TransformsCatalog catalog, params (string input, string output)[] columns)
```

Comment:
> (string input, string output)[] columns [](start = 96, length = 40)

how come some extensions are multi input/output with tuples, some are single input/output, and some are multi input/output with ColumnInfo?

Reply:
yeah, one per constructor; they follow the ctor signatures.

---
In reply to: [229707525](https://github.com/dotnet/machinelearning/pull/1460#discussion_r229707525) [](ancestors = 229707525)